glfw/src/Window.zig

```diff
@@ -205,66 +203,123 @@ pub const Hint = enum(c_int) {
     x11_instance_name = c.GLFW_X11_INSTANCE_NAME,
 };
 
-/// Sets the specified window hint to the desired value.
-///
-/// This function sets hints for the next call to glfw.Window.create. The hints, once set, retain
-/// their values until changed by a call to this function or glfw.window.defaultHints, or until the
-/// library is terminated.
-///
-/// This function does not check whether the specified hint values are valid. If you set hints to
-/// invalid values this will instead be reported by the next call to glfw.createWindow.
-///
-/// Some hints are platform specific. These may be set on any platform but they will only affect
-/// their specific platform. Other platforms will ignore them.
-///
-/// Possible errors include glfw.Error.NotInitialized and glfw.Error.InvalidEnum.
-///
-/// @pointer_lifetime in the event that value is of a str type, the specified string is copied before this function returns.
-///
-/// @thread_safety This function must only be called from the main thread.
-///
-/// see also: window_hints, glfw.Window.defaultHints
-pub inline fn hint(h: Hint, value: anytype) Error!void {
-    const value_type = @TypeOf(value);
-    const value_type_info: std.builtin.TypeInfo = @typeInfo(value_type);
-
-    switch (value_type_info) {
-        .Int, .ComptimeInt => {
-            c.glfwWindowHint(@enumToInt(h), @intCast(c_int, value));
-        },
-        .Bool => {
-            const int_value = @boolToInt(value);
-            c.glfwWindowHint(@enumToInt(h), @intCast(c_int, int_value));
-        },
-        .Enum => {
-            const int_value = @enumToInt(value);
-            c.glfwWindowHint(@enumToInt(h), @intCast(c_int, int_value));
-        },
-        .Array => |arr_type| {
-            if (arr_type.child != u8) {
-                @compileError("expected array of u8, got " ++ @typeName(arr_type));
-            }
-            c.glfwWindowHintString(@enumToInt(h), &value[0]);
-        },
-        .Pointer => |pointer_info| {
-            const pointed_type = @typeInfo(pointer_info.child);
-            switch (pointed_type) {
-                .Array => |arr_type| {
-                    if (arr_type.child != u8) {
-                        @compileError("expected pointer to array of u8, got " ++ @typeName(arr_type));
-                    }
-                },
-                else => @compileError("expected pointer to array, got " ++ @typeName(pointed_type)),
+/// Window hints
+pub const Hints = struct {
+    resizable: bool = true,
```

Comment:
```suggestion
    // Note: The defaults here are directly from the GLFW source of the glfwDefaultWindowHints function
    resizable: bool = true,
```

Reply:
Adding this note so that others will know how to verify if these values are correct or not and how to update them in the future